### ISSUE TITLE
passing an object {......, merge: true } to navigation.navigate() to enable merging of params is not supported by typescript

### ISSUE THREAD

**Issue by itsramiel** [react-navigation/react-navigation#10169]
### Current behavior

I have three screen in native stack navigation. I am passing a param eg: {numOfPosts: 5} to screen 2 from screen 1, and then from screen3 I want to go back to screen 2 without the need to pass the {numOfPosts:5} again.

I saw this in the docs to enable merging
https://reactnavigation.org/docs/upgrading-from-5.x/#params-are-now-overwritten-on-navigation-instead-of-merging

everything works well when writing navigation.navigate({name: "Screen2", mergeL true}), and there are no runtime errors, but I am getting a typescript compilation error

`property 'params' is missing in type '{ name: "AllPosts"; merge: true; }' but required in type '{ name: "AllPosts"; key?: string | undefined; params: { numOfPosts: number; }; merge?: boolean | undefined; }'`

### Expected behavior

I am expecting to not get a compilation error because it is valid code given from the docs.

### Reproduction

https://github.com/ramiel1999/react-navigation-merge-typescript

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [X] @react-navigation/native-stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
|@react-navigation/bottom-tabs| not used
|@react-navigation/drawer | not used
|@react-navigation/material-bottom-tabs | not used
|@react-navigation/material-top-tabs | not used
|@react-navigation/stack | not used
| @react-navigation/native               | ^6.0.6
| @react-navigation/native-stack         | ^6.2.5
| react-native-safe-area-context         | ^3.3.2
| react-native-screens                   |  ~3.8.0
| react-native                           | 0.64.3
| expo                                   | ^43.0.3

**Comment by uloco** [comment]
This pull request looks promising. Only one review is missing ....
https://github.com/react-navigation/react-navigation/pull/9982

**Comment by uloco** [comment]
Is there any way to type this correctly? Since the PR mentioned above has been closed due to some problems with the typings. Would be really great since we migrated from an old codebase and in a lot of places we have to merge the params now and everywhere we have to ignore the types. Basically whenever you want to navigate back to a screen we don't want to override the params, since it should be in cache...
